pkg/syfs/syfs.go

```diff
@@ -61,6 +62,15 @@ func configDir(dir string) string {
 	}
 
 	homedir := os.Getenv("HOME")
+	needUpdate := false
+	if homedir != "" && !fs.IsWritable(homedir) {
+		// apptainer process can not write the homedir, should ignore the manual value
+		sylog.Warningf("Environment variable `HOME` is set to a non-writable path:%s, will reset it", homedir)
+		// reset the homedir based on user's info and current working dir
+		homedir = ""
+		needUpdate = true
```

Comment:
I don't see why this is deferred instead of updating $HOME right there.

Reply:
thanks!, code is updated.